src/backends/postgresql/statement.cpp

```diff
@@ -783,7 +783,7 @@ void postgresql_statement_backend::describe_column(int colNum, data_type & type,
     {
         int form = PQfformat(result_, pos);
         int size = PQfsize(result_, pos);
-        if (form == 0 && size == -1)
+        if (form == 0 && (size == -1 || size == 4))
```

Comment:
Isn't it a bit dangerous to assume that any unknown OID of length 4 is a string? In my Postgres installation
```sql
select typname from pg_type where typlen=4;
```
returns ~30 types some of them don't look very stringy.

I think that for the proper enum support we really ought to query `pg_type` and check if `typcategory=='E'` (see https://www.postgresql.org/docs/current/catalog-pg-type.html#CATALOG-TYPCATEGORY-TABLE).